apps/package.json

```diff
@@ -170,6 +170,7 @@
   "dependencies": {
     "filesaver.js": "0.2.0",
     "jszip": "3.0.0",
-    "query-string": "4.1.0"
+    "query-string": "4.1.0",
+    "react-addons-update": "^15.3.0"
```

Comment:
Upon looking at the code further, it does look like a reasonable usage. It might also be worth a gander at immutable-js, which we use in a few places to do similar sorts of things.